core/src/main/scala/io/qbeast/spark/writer/RollupDataWriter.scala

```diff
@@ -100,9 +101,12 @@ trait RollupDataWriter extends DataWriter {
       val weight = Weight(extendedRow.getInt(qbeastColumns.weightColumnIndex))
       val cubeIdBytes = extendedRow.getBinary(qbeastColumns.cubeColumnIndex)
       val cubeId = revision.createCubeId(cubeIdBytes)
-      val rollupCubeIdBytes = extendedRow.getBinary(qbeastColumns.cubeToRollupColumnIndex)
-      val rollupCubeId = revision.createCubeId(rollupCubeIdBytes)
-      (row, weight, cubeId, rollupCubeId)
+      val filename = {
```

Comment:
The column with which the data was partitioned (line 58) and the value we organize index file writers with (line 84) must be the same. You can use an internal variable to sync them.

Reply:
Technically it is the same:

In line 58 I use:
```
.repartition(col(QbeastColumns.fileUUIDColumnName))
```
In line 84 we use filename, which is extracted from:
```
  val filename = {
    if (qbeastColumns.hasFilenameColumn)
      extendedRow.getString(qbeastColumns.filenameColumnIndex)
    else extendedRow.getString(qbeastColumns.fileUUIDColumnIndex) + ".parquet"
  }
```
Both contains the same UUID, the difference is that a table format can provide a custom filename (containing this UUID) or not. In case of delta the `qbeastColumns.filenameColumnIndex` column is not set, so it will use the `qbeastColumns.fileUUIDColumnIndex`. In case of hudi `qbeastColumns.filenameColumnIndex` is set, so it will use it (contains the UUID in `qbeastColumns.fileUUIDColumnName` + custom filename convention).